### PR TITLE
[fix](mtmv) Fix duplicate column name not check when create materialized view

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes
This is brought by https://github.com/apache/doris/pull/26146
If create materialized view as following, Should fail, because has the duplicated column name `o_orderdatE` and `o_orderdate`. But now can create materialized view successfully. the pr fix this.

```sql
        CREATE MATERIALIZED VIEW mv_1
        BUILD IMMEDIATE REFRESH AUTO ON MANUAL 
        partition by(o_orderdate) 
        DISTRIBUTED BY RANDOM BUCKETS 2 
        PROPERTIES ('replication_num' = '1') 
        AS  
        select o_orderdatE, o_shippriority, o_comment, o_orderdate, 
        sum(o_totalprice) as sum_total, 
        max(o_totalpricE) as max_total, 
        min(o_totalprice) as min_total, 
        count(*) as count_all, 
        bitmap_union(to_bitmap(case when o_shippriority > 1 and o_orderkey IN (1, 3) then o_custkey else null end)) cnt_1, 
        bitmap_union(to_bitmap(case when o_shippriority > 2 and o_orderkey IN (2) then o_custkey else null end)) as cnt_2 
        from (select * from orders) as t1
        group by 
        o_orderdatE, 
        o_shippriority, 
        o_comment,
        o_orderdate;
```

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java [hunk 1]
```diff
@@ -195,6 +197,16 @@ public void analyze(ConnectContext ctx) throws Exception {
         rewriteQuerySql(ctx);
     }
 
+    /**validate column name*/
+    public void validateColumns(List<ColumnDefinition> columns) throws UserException {
+        Set<String> colSets = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        for (ColumnDefinition col : columns) {
+            if (!colSets.add(col.getName())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java:204]: do we check invalid character in column name?

Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java:204]: Yeah,already have the logic to check invalid caracter in column name, and move the check invalid caracter logic to method `validateColumns`

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: regression-test/suites/mtmv_p0/same_column_name_check/same_column_name_check.groovy [hunk 2]
```diff
@@ -0,0 +1,102 @@
+package same_column_name_check
```
Comment (morrySnow) [inline regression-test/suites/mtmv_p0/same_column_name_check/same_column_name_check.groovy:1]: what's this?

Comment (seawinde) [reply regression-test/suites/mtmv_p0/same_column_name_check/same_column_name_check.groovy:1]: seems package name, auto genereated when create groovy package

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**